slides/aml-07-020718-preprocessing/index.html

```diff
@@ -0,0 +1,1094 @@
+      }
+      .largest, .largest .remark-code {
+          font-size: 50px;
+      }
+      .left-column, .right-column {
+          width: 48%;
+      }
+      .right-column{
+          float: right;
+      }
+      .left-column{
+          float: left;
+      }
+      .narrow-right-column {
+          float: right;
+          width: 32%
+      }
+      .wide-left-column {
+          float: left;
+          width: 65%
+      }
+      .invisible {
+          visibility: hidden
+      }
+      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
+    </style>
+  </head>
+  <body>
+    <textarea id="source">
+
+class: center, middle
+
+### W4995 Applied Machine Learning
+
+# Preprocessing and Feature Engineering
+
+02/08/17
+
+Andreas C. Müller
+
+???
+Today we’ll talk about preprocessing and featureengineering. What we’re talking about today mostly
+applies to linear models, and not to tree-based
+models, but it also applies to neural nets and kernel
+SVMs.
+
+---
+
+class: left, middle
+
+# Notes on homework
+
+- Next one will be shorter.
+- Good job everybody !
+- Tools are important !
+  - Don’t use your systems Python, use environments
+  - Pick a decent editor / IDE!
+  - If you’re on windows, consider dual booting
+
+???
+I realized the homework was a bit much for some of
+you who haven’t been familiar with git and testing.
+I saw many of you made really great progress and got
+everything together, so good on you. I promise the
+next homework will be shorter.
+While this might have been much all at once, all the
+pieces are important. Who wants to go to industry?
+You’ll fight these fights all the time!
+I gave some advice at the second lecture, and I
+noticed not everybody took it. If you want be a
+serious data scientist in industry, start now. Less
+applicable to researchers (but also).
+Don’t use system python. Know which environment
+you’re using. Pick an editor.
+If you’re on windows, get comfy with cygwin or better
+set up dual boot. You need to know POSIX!
+
+---
+
+class: left, middle
+
+.center[
+![:scale 90%](images/img_1.png)
+]
+
+
+???
+Let’s go back to the boston housing dataset. The idea
+was to predict house prices. Here are the features on
+the x axis and the response, so price, on the y axis.
+
+What are some thing you can notice? (concentrated
+distributions, skewed distibutions, discrete variable,
+linear and non-linear effects, different scales)
+
+---
+
+
+class: centre, middle
+
+#Scaling
+
+
+???
+N/A
+
+---
+
+class: centre, middle
+
+.center[
+![:scale 70%](images/img_2.png)
+]
+
+
+???
+Let’s start with the different scales.
+
+Many model want data that is on the same scale.
+KNearestNeighbors: If the distance in TAX is
+between 300 and 400 then the distance difference in
+CHArS doesn’t matter!
+
+Linear models: the different scales mean different
+penalty. L2 is the same for all!
+
+We can also see non-gaussian distributions here btw!
+---
+class: centre, middle
+
+#Ways to Scale Data
+
+.center[
```

Comment:
this is already in a center class because of the "class: center" above

Reply:
Actually, you wrote "centre" above, which I assume is a typo?

Reply:
Noted. And yes, its a typo.